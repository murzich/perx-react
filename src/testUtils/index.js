// eslint-disable-next-line import/prefer-default-export
export { default as renderWithRedux } from './renderWithRedux';
