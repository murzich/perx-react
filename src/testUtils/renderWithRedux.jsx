import { createStore } from 'redux';
import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import reducer from '../state/reducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
