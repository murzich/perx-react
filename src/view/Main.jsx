import React from 'react';
import Repos from './repos';
import Orgs from './orgs';
import SearchForm, { SearchProvider } from './userSearch';

const Main = () => (
  <main>
    <SearchProvider>
      <SearchForm />

      <Repos />

      <Orgs />
    </SearchProvider>
  </main>
);

export default Main;
