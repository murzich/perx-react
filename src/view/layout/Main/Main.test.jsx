import { cleanup } from '@testing-library/react';
import React from 'react';
import { renderWithRedux } from '../../../testUtils';
import Main from './Main';

describe('Main component', () => {
  afterAll(cleanup);

  const { getByText, getByLabelText, asFragment } = renderWithRedux(<Main />);

  it('should render title', () => {
    expect(getByText('GitHub users explorer')).toBeDefined();
  });

  it('should render the search input', () => {
    expect(getByLabelText('Search by username')).toBeDefined();
  });

  it('should render repositories section', () => {
    expect(getByText('Repositories')).toBeDefined();
  });

  it('should render organizations section', () => {
    expect(getByText('Organizations')).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
