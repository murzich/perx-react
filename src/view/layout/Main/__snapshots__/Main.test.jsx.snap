// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui container root"
  >
    <main
      class="ui segment"
    >
      <h1
        class="ui header"
      >
        GitHub users explorer
      </h1>
      <div
        class="ui hidden divider"
      />
      <form>
        <div
          class="ui action left icon input"
          width="6"
        >
          <input
            aria-label="Search by username"
            name="user"
            placeholder="Type user nameâ€¦"
            type="search"
          />
          <i
            aria-hidden="true"
            class="github icon"
          />
          <button
            class="ui button"
            type="submit"
          >
            search
          </button>
        </div>
      </form>
      <div
        class="ui divider"
      />
      <div
        class="ui stackable divided two column grid"
      >
        <div
          class="row"
        >
          <div
            class="column"
          >
            <section>
              <h2
                class="ui header"
              >
                Repositories
              </h2>
              <div
                class="ui divided relaxed list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                >
                  <div
                    class="content"
                  >
                    <div
                      class="header"
                    >
                      Enter user name to search for his repositories
                    </div>
                    <div
                      class="description"
                    >
                      There should be a list of all his repositories with public access.
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="column"
          >
            <section>
              <h2
                class="ui header"
              >
                Organizations
              </h2>
              <div
                class="ui divided relaxed list"
                role="list"
              >
                <div
                  class="item"
                  role="listitem"
                  style="display: flex; align-items: flex-start;"
                >
                  <div
                    class="ui placeholder"
                    style="height: 32px; width: 32px; margin-right: 1rem;"
                  >
                    <div
                      class="image"
                    />
                  </div>
                  <div
                    class="content"
                  >
                    <div
                      class="header"
                    >
                      Enter user name to search for organizations
                    </div>
                    <div
                      class="description"
                    >
                      There should be a list of all his organizations with public access.
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
